### PR TITLE
Refresh Package before Post + move few PHPMD rules to PHP_CodeSniffer

### PR DESCRIPTION
**Pull request by TomasVotruba**
- try on recent PHP versions
- bump composer dependencies
- move tests to standard `/tests` directory; seperate code and tests also prevent downloading tests while used as dependency
- move src to standard `/src` directory; respects PSR-4 with no redundant directory nesting; also 1:1 to tests

As agreed with @vitek-rostislav 

### TODO 

- [x] make it easier for Windows users to run the tools
- [x] cleanup docs to not contain duplicate information that can be found in rulesets or docs of the tools
- <strike> keep documented why are certain rules disabled (see https://github.com/shopsys/coding-standards/pull/6#discussion_r158069658)</strike>
- [x] merge `<!-- From php-md -->` and `<!-- Custom ShopSys standards -->` rules lists (see https://github.com/shopsys/coding-standards/pull/6#discussion_r158479825)
### In next PR
- keep `private $_varName` as a prohibited property name (see https://github.com/shopsys/coding-standards/pull/6/commits/f406eb26ef769219ea9023a58b3f985a2099994f#r158484030)

### CODE REVIEW

File: rulesetMD.xml
Comment (sustmi) [inline rulesetMD.xml]: The purpose of this is to document why we do not use the rule. How else would you recommend to overcome the situation that somebody will spend unnecessary time trying to enable the rule and figuring out that it cannot be used and disabling it again (over and over)?

File: .travis.yml
Comment (sustmi) [inline .travis.yml]: The script is prefixed with `php` because it is then easier for Windows CMD users to simply copy and run the command.
In Windows CMD running `vendor/bin/parallel-lint ./src` would not work. You need Unix shell that understands   [Shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) for this. This could lead to unnecessary confusion.

Too bad it is not included in `README.md` too. It was probably written by somebody else who did not realize this. :/

File: rulesetCS.xml
Comment (sustmi) [inline rulesetCS.xml]: By the way: What is the rationale behind moving checks from PHPMD to PHP-CS-Fixer?

File: docs/description-of-used-coding-standards-rules.md
Comment (sustmi) [inline docs/description-of-used-coding-standards-rules.md]: To be honest, I am not really sure if the file has adequate value. It basically duplicates the information already present in different files (something like `SELECT name, description FROM our_ruleset JOIN tool_documentation`) but we already have it and that means we should maintain it.
You removed mentions of PHPMD rules but did not add appropriate PHP-CS-Fixer rules. Can you please fix it?

**Review by sustmi — CHANGES_REQUESTED** [review]
Hi @TomasVotruba . Thank you for your contribution!
I added few minor comments. Please fix the issues / answer the comments so that we can merge your code.

File: rulesetCS.xml
Comment (sustmi) [reply rulesetCS.xml]: Oh, I can see now: https://github.com/shopsys/coding-standards/issues/7
You want to completely replace PHPMD.

File: .travis.yml
Comment (TomasVotruba) [reply .travis.yml]: Oh, I didn't know that. I thought this is now standard on Windows as well.

Do you think that  Windows users would be confused by that? I imagine they're used to it and know what you've just written. So they prefix commands by default.

I suppose they already had to use comopser, so seing `composer` would mean `php composer` for them. Or am I wrong?

File: docs/description-of-used-coding-standards-rules.md
Comment (TomasVotruba) [reply docs/description-of-used-coding-standards-rules.md]: > To be honest, I am not really sure if this file has an adequate value. It basically duplicates the information already present in different files

I agree it has barely any value. Rules are already clear from configs, they should be commented there if you need to add something extra, like [this in this config](https://github.com/Symplify/Symplify/blob/master/packages/EasyCodingStandard/config/clean-code.neon).

Also, you package should document **your code**. This is 3rd party code that is already covered on their side.

**I suggest removing this one and only document your extra Sniffs/Fixers.**

File: rulesetMD.xml
Comment (TomasVotruba) [reply rulesetMD.xml]: I guess this is already covered in #7 

Also these *WIP code* should be documented in issues, otherwise it's only dead code and that should never be part of `master` branch of open-source project.

File: .travis.yml
Comment (TomasVotruba) [reply .travis.yml]: In `travis.yml` it doesn't make sense to add these, since it *nix environment only.

I could revert that in `README.md` if you think Windows users are not aware of it.

What do you think?

**Comment by TomasVotruba** [comment]
@sustmi Thanks for you feedback.

Based on it and on my developer experience with this packages, I've proposed more informative `README`.

Let me know what you think :+1:

File: rulesetCS.xml
Comment (sustmi) [reply rulesetCS.xml]: One more thing: I do not see any reason to put the fixers that were PHPMD rules before in separate section called `<!-- From php-md -->`. For anybody in future it is irrelevant where were the rules before.

It is good for a reviewer like me to realize what happened but this is the kind of information that belongs into commit message.

Can you please merge the items from `<!-- From php-md -->` int to the `<!-- Custom ShopSys standards -->` section? Also note that the rules are kept in alphabetical order (for better clarity).

File: rulesetCS.xml
Comment (sustmi) [inline rulesetCS.xml]: This rule allows:
```php
private $_varName;
```
which `CamelCaseVariableName` did not allow.

File: rulesetMD.xml
Comment (sustmi) [reply rulesetMD.xml]: I am not sure if you understand. If you are only one maintainer you probably remember which rules you enabled and which not and why.

But somebody else might see this ruleset and say "hey, why don't you use *ElseExpression* rule"? And I will look at it and say "hmm, seems like a good idea, lets enable it". Then I will enable it and see why we did not enable it and say to myself "oh, that's why". This already happened. I did not keep the rules commented for no reason.

Also, this is not a *dead code*. It's in-line documentation of reason why we do not enable the rule. And it's not a *WIP code* too because we actually do not plan to rewrite the code in the future to adhere these rules.

Comment (sustmi) [reply rulesetMD.xml]: We could document the reasons for not enabling it in docs as well but I think that the ruleset file it the best place for it because this is the only place where you are sure that somebody will look when trying to enable some rule (people do not expect to find such information in documentation).

Well, maybe when this repository is all about coding standards then *docs* directory is a good place for this (but in case of a ruleset file in a normal project I would definitely put the info directly in the ruleset file).

File: .travis.yml
Comment (sustmi) [reply .travis.yml]: Actually, Composer installer for Windows adds `composer.bat` to your `PATH` file that makes the command work without the `php` prefix.

The `php` prefix you added in "README: add Windows copy-paste support" commit solves this for me. 👍 

(Even though the commit message could be a bit misleading. Something like "README: make commands platform independent" would be better.)

File: docs/description-of-used-coding-standards-rules.md
Comment (sustmi) [reply docs/description-of-used-coding-standards-rules.md]: I agree. Lets keep only the information that cannot be found somewhere else. You did a good job removing the file completely in https://github.com/shopsys/coding-standards/pull/6/commits/ac9212c5c5517efada4a59780f10265fb6b59514.

The new texts in the README will need some minor fixes but I will do it after we merge this. It's a pity that I as a reviewer cannot somehow add commits into your PR - that would make it much easier.

**Comment by sustmi** [comment]
Hi @TomasVotruba . There are few last steps for this PR to be good to merge. GitHub's "review" feature does not make it easy to realize which comments were solved and which not so added a TODO list into PR description.

Also, please rebase the PR to the current master. I found some minor issue in `OrmJoinColumnRequireNullableFixer.php` and fixed it and as you are moving the file in this PR it creates a conflict now.

Thanks.

File: .travis.yml
Comment (TomasVotruba) [reply .travis.yml]: (I probably tried to include the motivation behind the commit to the message. I agree "platform independ" sounds more proffesional, will use that next time).

This is resolve then?

File: rulesetCS.xml
Comment (TomasVotruba) [reply rulesetCS.xml]: You are right. I agree!

File: rulesetMD.xml
Comment (TomasVotruba) [reply rulesetMD.xml]: I understand you've been discussion this internally and there is more work then meets the eye.


I think **end user** like me will ask **3 important questions**:

- 1. **Do you use PSR-2 set?**
- 2. **What extra do you offer?**
- 3. Both conclude to: **Why should I use this package over clean PHP_CodeSniffer with own ruleset?**

Motivation of chosen rules or not-chosen rules is not so relevant for majority of users. I'll mention it again, I didn't noticed this file until I send PR. That should not happen. User should read everything important about this right from the `README.md`.

I'd invest into answering these 3 question rather then to describing known rules.

**Comment by TomasVotruba** [comment]
Hey @sustmi , thanks for the extensive review. It gives me more insights on how you work with repositories on Github.

As for managining *todos*, I have a tip for you. For resolved comments is use :+1:  on them. When I mark comment as :+1: , it is resolved to continue or finished. I use it over few Github organizations, it's very easy to see and also for others to follow.


Rebased :white_check_mark:

**Comment by TomasVotruba** [comment]
As for:

> keep private $_varName as a prohibited property name (see f406eb2#r158484030)

I recommend resolving this in another PR. It's already grown too much out of scope of original idea. Extending it with more big changes like custom fixers would make it one huge refactoring PR. Also @vitek-rostislav is waiting for this PR to merge with his blog post.

File: .travis.yml
Comment (sustmi) [reply .travis.yml]: Yes it is.

**Pull request merged**
